service/gcs/oslayer/realos/realos.go

```diff
@@ -184,6 +185,18 @@ func (o *realOS) Unmount(target string, flags int) (err error) {
 	}
 	return nil
 }
+func (o *realOS) UnplugSCSIDisk(scsiID string) (err error) {
+	f, err := os.OpenFile(filepath.Join("/sys/bus/scsi/devices", scsiID, "delete"), os.O_WRONLY, 0644)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	defer f.Close()
+
+	if _, err := f.Write([]byte("1\n")); err != nil {
```

Comment:
Just trying to get my head around what this does - writing `1` here? But LGTM otherwise.

Reply:
the files in `/sys` are magic :) For some of them if you write anything to them stuff will happen. For the `delete` file in the `scsi` hierarchy writing anything to the file will case the unplugging. It is customary to do something like `echo 1 > /sys/..../delete`, which basically this code does.